onyo/lib/onyo.py

```diff
@@ -1240,3 +1241,128 @@ def unset(self, paths: Iterable[Union[Path, str]], values: str,
             self.restore()
 
         return diff
+
+    #
+    # GET
+    #
+    # TODO: This should be a staticmethod in utils
+    def _walk(self, path, depth) -> Generator:
+        """Recursively yield filepaths up to a given depth"""
+        depth -= 1
+        with os.scandir(path) as p:
+            for entry in p:
+                # ignore hidden files and folders
+                # TODO: Better to use _is_protected_path or ignore hidden?
+                #   `if self._is_protected_path(entry.path):`
+                if entry.name.startswith('.'):
+                    continue
+
+                # yield discovered assets
+                if not entry.is_dir():
+                    yield entry.path
+
+                # explore dirs if depth allows
+                if entry.is_dir() and depth >= 0:
+                    yield from self._walk(entry.path, depth)
+
+    @staticmethod
+    def _filter_by_name(assets: List[str], filters: dict) -> Generator:
+        """Filter assets by filename to prevent redundant asset loading"""
```

Comment:
How would there be redundant assets in the asset list? `self.assets` is a `set`.

Reply:
The use of redundant here is "no longer useful/needed". `self.assets` contains all assets, so we filter out assets first by name because it contains type, make, model, and serial data. Only then do we pass the asset list to more stringent filtering that requires the asset to be loaded.

Reply:
So you're saying the docstring should be:
```
"""Filter assets by filename"""
```

Correct?